node/p2p_block_sync.md

```diff
@@ -0,0 +1,84 @@
+    participant Full Node 1
+    participant Full Node 2
+    participant DA Layer
+
+    User->>Block Producer: Send Tx
+    Block Producer->>Block Producer: Generate Block
+    Block Producer->>DA Layer: Publish Block
+
+    Block Producer->>Full Node 1: Gossip Block
+    Block Producer->>Full Node 2: Gossip Block
+    Full Node 1->>Full Node 1: Verify Block
+    Full Node 1->>Full Node 2: Gossip Block
+    Full Node 1->>Full Node 1: Mark Block Soft-Confirmed
+
+    Full Node 2->>Full Node 2: Verify Block
+    Full Node 2->>Full Node 2: Mark Block Soft-Confirmed
+
+    DA Layer->>Full Node 1: Retrieve Block
+    Full Node 1->>Full Node 1: Mark Block Hard-Confirmed
+
+    DA Layer->>Full Node 2: Retrieve Block
+    Full Node 2->>Full Node 2: Mark Block Hard-Confirmed
+```
+
+## Protocol/Component Description
+
+P2P Block Sync consists of the following components:
+
+* block exchange service: responsible for gossiping blocks over P2P
+* block publication to P2P network
+* block retrieval from P2P network
+
+### Block Exchange Service
+
+The block exchange service is created during full node initialization. After that, during the block manager's initialization, a pointer to the block store inside block exchange service is passed to it. Blocks created in the block manager which need to gossipped over a p2p network are then passed to the `BlockCh` channel and then sent to [go-header] service to be gossip blocks over P2P network.
+
+### Block Publication to P2P network
+
+Blocks ready to be published to the P2P network are sent to the `BlockCh` channel in Block Manager inside `publishLoop`.
+The `blockPublishLoop` in the full node continuously listens for new blocks from the `BlockCh` channel and when a new block  
+is received, it is written to the block store and broadcasted to the network using the block exchange service.
+
+### Block Retrieval from P2P network
+
+Blocks gossiped to validating full nodes through the P2P network are retreived from the `Block Store` in `BlockStoreRetrieveLoop` in Block Manager.
+For every `blockTime` unit of time, a signal is sent to the `blockStoreCh` channel in block manager and when this signal is received, the  
+`BlockStoreRetrieveLoop` retrieves blocks from the block store. It keeps track of the last retrieved block's height and if the current block store's height  is greater than the last retrieved block's height, it retrieves all blocks from the block store that are between these two heights.
+For each retrieved block, it sends a new block event to the `blockInCh` channel which is the same channel that blocks retrieved from the DA layer are sent.
+This block is marked as soft-confirmed by the validating full node until the same block is seen on the DA layer and then marked hard-confirmed.
+
+## Message Structure/Communication Format
+
+The communication within Block Manager and between itself and the full node is all done through channels that pass around the `block struct`.  
```

Comment:
we should have a link here to the block struct . Probably something like proposed here https://github.com/rollkit/rollkit/pull/1202#discussion_r1337517287

Reply:
Added